tests/multi_net/udp_recv_dontwait.py

```diff
@@ -0,0 +1,55 @@
+# Test UDP recv and recvfrom with MSG_DONTWAIT
+import random
+import socket
+
+try:
+    import errno, time
+
+    if not hasattr(time, "ticks_ms"):
+        raise ImportError  # skip the test for minimal unix ports w/o ticks_ms()
+except ImportError:
+    print("SKIP")
+    raise SystemExit
+
+
+# Server
+def instance0():
+    PORT = random.randrange(10000, 50000)
+    multitest.globals(IP=multitest.get_network_ip(), PORT=PORT)
+    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    s.bind(socket.getaddrinfo("0.0.0.0", PORT)[0][-1])
+    multitest.next()
+    begin = time.ticks_ms()
+
+    # do some recvs before instance1 starts, when we know no packet is waiting
+    try:
+        print(s.recvfrom(8, socket.MSG_DONTWAIT))
+    except OSError as e:
+        print(e.errno == errno.EAGAIN)
+    try:
+        print(s.recv(8, socket.MSG_DONTWAIT))
+    except OSError as e:
+        print(e.errno == errno.EAGAIN)
+
+    # the above steps should not have taken any substantial time
+    elapsed = time.ticks_diff(time.ticks_ms(), begin)
+    print(True if elapsed < 50 else elapsed)
+
+    # Now instance1 will send us a UDP packet
+    multitest.broadcast("0-ready")
+    multitest.wait("1-sent")
+    time.sleep(2)  # allow some extra time for packet to arrive
```

Comment:
This seems like it shouldn't be necessary, but it is...

Reply:
I can understand that this delay is necessary, to let the packet go through the network and be processed by the target device.

But I can reduce this to 0.5 and the tests still run reliably in both directions on RPI_PICO_W and RPI_PICO2_W.  Reducing it down to 0.1 and the tests are less reliable.

Alternatively, could wrap the `s.recv` below in a loop and retry if it raises EAGAIN?

Reply:
Ah, interesting again - anything less than 2 seconds fails on my setup

Reply:
Done